compendium/ServiceComponent/test/CMakeLists.txt

```diff
@@ -150,3 +153,12 @@ set_tests_properties(usServiceComponentInvalidInheritanceTest PROPERTIES PASS_RE
 set_tests_properties(usServiceComponentNoDepsNoDefaultCtorTest PROPERTIES PASS_REGULAR_EXPRESSION ".*An appropriate constructor was not found and/or the service implementation does not implement all of the service interface's methods. A default constructor or a constructor with an AnyMap input parameter for the configuration properties is required when inject-references is false..*")
 set_tests_properties(usServiceComponentNoDefaultCtorTest PROPERTIES PASS_REGULAR_EXPRESSION ".*An appropriate constructor was not found and/or the service implementation does not implement all of the service interface's methods. A constructor with service reference input parameters or a constructor with an AnyMap input parameter for configuration properties and service reference input parameters is required when inject-references is true.*")
 set_tests_properties(usServiceComponentRefCountTest PROPERTIES PASS_REGULAR_EXPRESSION ".*An appropriate constructor was not found and/or the service implementation does not implement all of the service interface's methods. A constructor with service reference input parameters or a constructor with an AnyMap input parameter for configuration properties and service reference input parameters is required when inject-references is true.*")
+if (US_COMPILER_MSVC)
```

Comment:
instead of tightly coupling the test to a compiler version/vendor, can custom strings be added to the `static_asserts` in [UserDefinedMethodAssertion.hpp](https://github.com/CppMicroServices/CppMicroServices/pull/1046/files#diff-801b07fb2f80837cf5fbea83fb8771f044d793cbbc14a3df09493592c1b8bb0e) and those checked instead of the compiler's? This would make the test not brittle to compiler upgrades/changes.

Reply:
Struggling to make this work as i don't believe the static assert is actually what ends up throwing. What ends up throwing is that attempt to call a modified method which does not exist and therefore is beholden to the compiler warning for that:

```19: Z:\compendium\ServiceComponent\include\cppmicroservices/servicecomponent/UserDefinedMethodAssertion.hpp(46,55): error C2039: 'Modified': is not a member of '`anonymous-namespace'::ServiceComponentWrongModifiedSig' [Z:\build\config_0\compendium\ServiceComponent\test\usServiceComponentModifiedMethod.vcxproj]
19: Z:\compendium\ServiceComponent\test\TestCompInst_IncorrectModifiedMethod.cpp(43): message : see declaration of '`anonymous-namespace'::ServiceComponentWrongModifiedSig' [Z:\build\config_0\compendium\ServiceComponent\test\usServiceComponentModifiedMethod.vcxproj]
19: Z:\compendium\ServiceComponent\include\cppmicroservices/servicecomponent/UserDefinedMethodAssertion.hpp(38): message : while compiling class template member function 'cppmicroservices::service::component::UserDefinedMethodAssertion<`anonymous-namespace'::ServiceComponentWrongModifiedSig>::~UserDefinedMethodAssertion(void)' [Z:\build\config_0\compendium\ServiceComponent\test\usServiceComponentModifiedMethod.vcxproj]
19: Z:\compendium\ServiceComponent\test\TestCompInst_IncorrectModifiedMethod.cpp(46): message : see reference to class template instantiation 'cppmicroservices::service::component::UserDefinedMethodAssertion<`anonymous-namespace'::ServiceComponentWrongModifiedSig>' being compiled [Z:\build\config_0\compendium\ServiceComponent\test\usServiceComponentModifiedMethod.vcxproj]
19: Z:\compendium\ServiceComponent\include\cppmicroservices/servicecomponent/UserDefinedMethodAssertion.hpp(47): error C2062: type 'unknown-type' unexpected [Z:\build\config_0\compendium\ServiceComponent\test\usServiceComponentModifiedMethod.vcxproj]```

Looking into how we can make this work...

Reply:
Using https://cppinsights.io/ may help  you get some insight into what the instantiated template code is that the compiler actually sees and tries to compile.